### PR TITLE
Fixed Modal Display and Sidebar Issues as Highlighted by David, HNG 9 Frontend PM for Zurichat

### PR DESCRIPTION
**Pull request by tryraisins**
# Fixes Issue/Linear Ticket

**My PR Fixes The Modal Display and Sidebar Issues as Highlighted by David, HNG 9 Frontend PM for zurichat**

# Changes proposed
Resolve Z-index Conflicts, Sidebar Resize Issues and Sidebar Dropdown Menu not Closing when clicking Outside Element 
## What were you told to do?
Fix Edit Workspace Details Modal, Can't close modal after opening

## What did you do?
Removed Z-index property in modalCon class in [ModalComponentStyles.module.css](https://github.com/zurichat/zc_main/blob/dev/packages/main/src/components/protected/sidebar/styles/ModalComponentStyles.module.css)
Removed Z-index property in orgDiv class in [Sidebar.module.css](https://github.com/zurichat/zc_main/blob/dev/packages/main/src/components/protected/sidebar/styles/Sidebar.module.css)
Reduced minimum width before sidebar automatically collapses in [Sidebar.jsx](https://github.com/zurichat/zc_main/blob/dev/packages/main/src/components/protected/sidebar/components/Sidebar.jsx)
Implemented useRef and added closemodal function in [Header.jsx](https://github.com/zurichat/zc_main/blob/dev/packages/main/src/components/protected/sidebar/components/Header.jsx)

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

https://user-images.githubusercontent.com/78821732/207362777-1557d455-d463-41f3-a139-2e396fe65254.mp4

### CODE REVIEW

File: packages/root-config/src/assets/plugin.importmap.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "@zuri/zuri-plugin-tools": "https://externaltools.zuri.chat/zuri-zuri-plugin-tools.js",
     "@zuri/zuri-plugin-channels": "https://channels.zuri.chat/static/main/dist/zuri-zuri-plugin-channels.js",
     "@zuri/zuri-plugin-contributions": "https://ct.zuri.chat/js/app.js",
-    "@zuri/zuri-plugin-messaging": "https://chat.zuri.chat/zuri-messaging.js",
+    "@zuri/zuri-plugin-messaging": "//localhost:5013/zuri-messaging.js",
```
Comment (JulianaSau) [inline packages/root-config/src/assets/plugin.importmap.json:17]: Change this back to the live link

File: packages/main/src/components/protected/sidebar/styles/Sidebar.module.css [hunk 2]
```diff
@@ -67,7 +67,7 @@
   padding: 2px 10px;
   background-color: #00b87c;
   color: #fff;
-  z-index: 0;
+  /* z-index: 0; */
```
Comment (JulianaSau) [inline packages/main/src/components/protected/sidebar/styles/Sidebar.module.css:70]: Remove this if it's not being used

File: packages/main/src/components/protected/sidebar/styles/ModalComponentStyles.module.css [hunk 3]
```diff
@@ -10,7 +10,7 @@
   border: 1px solid #bebebe;
   border-radius: 5px;
   /* padding: 10px 0; */
-  z-index: 9999;
+  /* z-index: 9999; */
```
Comment (JulianaSau) [inline packages/main/src/components/protected/sidebar/styles/ModalComponentStyles.module.css:13]: Remove this if it's not being used

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: packages/root-config/src/assets/plugin.importmap.json (hunk 1 above)
Comment (tryraisins) [reply packages/root-config/src/assets/plugin.importmap.json:17]: Alright

File: packages/main/src/components/protected/sidebar/styles/ModalComponentStyles.module.css (hunk 3 above)
Comment (tryraisins) [reply packages/main/src/components/protected/sidebar/styles/ModalComponentStyles.module.css]: All changes have been made

**Review by JulianaSau — APPROVED** [review]

**Pull request merged**